worlds/wargroove2/__init__.py

```diff
@@ -0,0 +1,194 @@
+        "English",
+        "wargroove2_en.md",
+        "wargroove2/en",
+        ["Fly Sniper"]
+    )]
+
+    options_presets = wargroove2_option_presets
+
+
+class Wargroove2World(World):
+    """
+    Command an army, in the sequel to the hit turn based strategy game Wargroove!
+    """
+    options: Wargroove2Options
+    options_dataclass = Wargroove2Options
+    settings: typing.ClassVar[Wargroove2Settings]
+    game = "Wargroove 2"
+    topology_present = True
+    web = Wargroove2Web()
+    level_list: [Wargroove2Level] = None
+    first_level: Wargroove2Level = None
+    final_levels: [Wargroove2Level] = None
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = location_table
+
+    def _get_slot_data(self):
+        return {
+            'seed': "".join(
+                self.random.choice(string.ascii_letters) for i in range(16)),
+            'income_boost': self.options.income_boost.value,
+            'commander_defense_boost': self.options.commander_defense_boost.value,
+            'starting_groove_multiplier': self.options.groove_boost.value,
+            'level_shuffle_seed': self.options.level_shuffle_seed.value,
+            'can_choose_commander': self.options.commander_choice.value != 0,
+            'final_levels': self.options.final_levels.value,
+            'death_link': self.options.death_link.value == 1
+        }
+
+    def generate_early(self):
+        # First level
+        self.first_level = get_first_level(self.player)
+
+        # Standard levels
+        self.level_list = get_level_table(self.player)
+        low_victory_checks_levels = list(level for level in self.level_list if level.low_victory_checks)
+        high_victory_checks_levels = list(level for level in self.level_list if not level.low_victory_checks)
+        if self.options.level_shuffle_seed == 0:
+            random = self.random
+        else:
+            random = Random(str(self.options.level_shuffle_seed))
+
+        random.shuffle(low_victory_checks_levels)
+        random.shuffle(high_victory_checks_levels)
+        non_starting_levels = high_victory_checks_levels + low_victory_checks_levels[4:]
+        random.shuffle(non_starting_levels)
+        self.level_list = low_victory_checks_levels[0:4] + non_starting_levels
+
+        # Final Levels
+        self.final_levels = get_final_levels(self.player)
+        final_levels_no_ocean = list(level for level in self.final_levels if not level.has_ocean)
+        final_levels_ocean = list(level for level in self.final_levels if level.has_ocean)
+        random.shuffle(final_levels_no_ocean)
+        random.shuffle(final_levels_ocean)
+        non_north_levels = final_levels_ocean + final_levels_no_ocean[1:]
+        random.shuffle(non_north_levels)
+        self.final_levels = final_levels_no_ocean[0:1] + non_north_levels
+
+        # Selecting a random starting faction
+        if self.options.commander_choice == 2:
```

Comment:
```suggestion
        if self.options.commander_choice == "random_starting_faction":
```
Would recommend either comparing to the string or the option class instead of using what is basically a magic number. Makes it easier to read and debug.
For the option class, you'd do:
```py
from .Options import CommanderChoice

if self.options.commander_choice == CommanderChoice.option_random_starting_faction:
```
Both of these are accepted by core, comparing to a number (except in some cases like if it's a range or something) is generally not accepted and you'll get a comment on it later on if you do that.